// workbox-recipes/index.js
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/workbox-recipes/index.js
# version: v28.0.0
# last_commit: '2025-09-23T07:20:25+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
/*
  Copyright 2020 Google LLC

  Use of this source code is governed by an MIT-style
  license that can be found in the LICENSE file or at
  https://opensource.org/licenses/MIT.
*/
import { googleFontsCache } from './googleFontsCache';
import { imageCache } from './imageCache';
import { staticResourceCache, } from './staticResourceCache';
import { pageCache } from './pageCache';
import { offlineFallback } from './offlineFallback';
import { warmStrategyCache } from './warmStrategyCache';
import './_version.js';
/**
 * @module workbox-recipes
 */
export { googleFontsCache, imageCache, offlineFallback, pageCache, staticResourceCache, warmStrategyCache, };
