// eslint-plugin-jsx-a11y/__mocks__/JSXElementMock.js
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/eslint-plugin-jsx-a11y/__mocks__/JSXElementMock.js
# version: v20.0.0
# last_commit: '2025-09-25T18:50:22+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
/**
 * @flow
 */

import type { JSXAttributeMockType } from './JSXAttributeMock';

export type JSXElementMockType = {
  type: 'JSXElement',
  openingElement: {
    type: 'JSXOpeningElement',
    name: {
      type: 'JSXIdentifier',
      name: string,
    },
    attributes: Array<JSXAttributeMockType>,
  },
  children: Array<Node>,
};

export default function JSXElementMock(
  tagName: string,
  attributes: Array<JSXAttributeMockType> = [],
  children?: Array<Node> = [],
): JSXElementMockType {
  return {
    type: 'JSXElement',
    openingElement: {
      type: 'JSXOpeningElement',
      name: {
        type: 'JSXIdentifier',
        name: tagName,
      },
      attributes,
    },
    children,
  };
}
