// @emotion/babel-plugin/src/emotion-macro.js
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/@emotion/babel-plugin/src/emotion-macro.js
# version: v28.0.0
# last_commit: '2025-09-23T07:20:25+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
import { transformExpressionWithStyles, createTransformerMacro } from './utils'

const isAlreadyTranspiled = path => {
  if (!path.isCallExpression()) {
    return false
  }

  const firstArgPath = path.get('arguments.0')

  if (!firstArgPath) {
    return false
  }

  if (!firstArgPath.isConditionalExpression()) {
    return false
  }

  const alternatePath = firstArgPath.get('alternate')

  if (!alternatePath.isObjectExpression()) {
    return false
  }

  const properties = new Set(
    alternatePath.get('properties').map(p => p.node.key.name)
  )

  return ['name', 'styles'].every(p => properties.has(p))
}

let createEmotionTransformer =
  (isPure /*: boolean */) =>
  (
    { state, babel, importSource, reference, importSpecifierName } /*: Object */
  ) => {
    const path = reference.parentPath

    if (isAlreadyTranspiled(path)) {
      return
    }

    if (isPure) {
      path.addComment('leading', '#__PURE__')
    }

    let node = transformExpressionWithStyles({
      babel,
      state,
      path,
      shouldLabel: true
    })
    if (node) {
      path.node.arguments[0] = node
    }
  }

export let transformers = {
  css: createEmotionTransformer(true),
  injectGlobal: createEmotionTransformer(false),
  keyframes: createEmotionTransformer(true)
}

export let createEmotionMacro = (importSource /*: string */) =>
  createTransformerMacro(transformers, { importSource })
