// eslint-plugin-react/lib/util/propWrapper.js
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/eslint-plugin-react/lib/util/propWrapper.js
# version: v20.0.0
# last_commit: '2025-09-23T07:20:25+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
/**
 * @fileoverview Utility functions for propWrapperFunctions setting
 */

'use strict';

const filter = require('es-iterator-helpers/Iterator.prototype.filter');
const some = require('es-iterator-helpers/Iterator.prototype.some');

function searchPropWrapperFunctions(name, propWrapperFunctions) {
  const splitName = name.split('.');
  return some(propWrapperFunctions.values(), (func) => {
    if (splitName.length === 2 && func.object === splitName[0] && func.property === splitName[1]) {
      return true;
    }
    return name === func || func.property === name;
  });
}

function getPropWrapperFunctions(context) {
  return new Set(context.settings.propWrapperFunctions || []);
}

function isPropWrapperFunction(context, name) {
  if (typeof name !== 'string') {
    return false;
  }
  const propWrapperFunctions = getPropWrapperFunctions(context);
  return searchPropWrapperFunctions(name, propWrapperFunctions);
}

function getExactPropWrapperFunctions(context) {
  const propWrapperFunctions = getPropWrapperFunctions(context);
  const exactPropWrappers = filter(propWrapperFunctions.values(), (func) => func.exact === true);
  return new Set(exactPropWrappers);
}

function isExactPropWrapperFunction(context, name) {
  const exactPropWrappers = getExactPropWrapperFunctions(context);
  return searchPropWrapperFunctions(name, exactPropWrappers);
}

function formatPropWrapperFunctions(propWrapperFunctions) {
  return Array.from(propWrapperFunctions, (func) => {
    if (func.object && func.property) {
      return `'${func.object}.${func.property}'`;
    }
    if (func.property) {
      return `'${func.property}'`;
    }
    return `'${func}'`;
  }).join(', ');
}

module.exports = {
  formatPropWrapperFunctions,
  getExactPropWrapperFunctions,
  getPropWrapperFunctions,
  isExactPropWrapperFunction,
  isPropWrapperFunction,
};
