// jsdom/lib/jsdom/living/range/AbstractRange-impl.js
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/jsdom/lib/jsdom/living/range/AbstractRange-impl.js
# version: v28.0.0
# last_commit: '2025-09-23T07:20:25+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
"use strict";

// https://dom.spec.whatwg.org/#abstractrange
class AbstractRangeImpl {
  constructor(globalObject, args, privateData) {
    const { start, end } = privateData;

    this._start = start;
    this._end = end;

    this._globalObject = globalObject;
  }

  // https://dom.spec.whatwg.org/#dom-range-startcontainer
  get startContainer() {
    return this._start.node;
  }

  // https://dom.spec.whatwg.org/#dom-range-startoffset
  get startOffset() {
    return this._start.offset;
  }

  // https://dom.spec.whatwg.org/#dom-range-endcontainer
  get endContainer() {
    return this._end.node;
  }

  // https://dom.spec.whatwg.org/#dom-range-endoffset
  get endOffset() {
    return this._end.offset;
  }

  // https://dom.spec.whatwg.org/#dom-range-collapsed
  get collapsed() {
    const { _start, _end } = this;
    return _start.node === _end.node && _start.offset === _end.offset;
  }
}

module.exports = {
  implementation: AbstractRangeImpl
};
