        <!-- POKERTOOL-HEADER-START
        ---
        schema: pokerheader.v1
project: pokertool
file: pokertool-frontend/node_modules/@typescript-eslint/eslint-plugin/docs/rules/consistent-type-definitions.md
version: v28.0.0
last_commit: '2025-09-25T18:50:22+01:00'
fixes:
- date: '2025-09-25'
  summary: Enhanced enterprise documentation and comprehensive unit tests added
        ---
        POKERTOOL-HEADER-END -->
---
description: 'Enforce type definitions to consistently use either `interface` or `type`.'
---

> üõë This file is source code, not the primary documentation location! üõë
>
> See **https://typescript-eslint.io/rules/consistent-type-definitions** for documentation.

TypeScript provides two common ways to define an object type: `interface` and `type`.

```ts
// type alias
type T1 = {
  a: string;
  b: number;
};

// interface keyword
interface T2 {
  a: string;
  b: number;
}
```

The two are generally very similar, and can often be used interchangeably.
Using the same type declaration style consistently helps with code readability.

## Options

- `"interface"` _(default)_: enforce using `interface`s for object type definitions.
- `"type"`: enforce using `type`s for object type definitions.

### `interface`

<!--tabs-->

#### ‚ùå Incorrect

```ts
/* eslint @typescript-eslint/consistent-type-definitions: ["error", "interface"] */

type T = { x: number };
```

#### ‚úÖ Correct

```ts
/* eslint @typescript-eslint/consistent-type-definitions: ["error", "interface"] */

type T = string;
type Foo = string | {};

interface T {
  x: number;
}
```

### `type`

<!--tabs-->

#### ‚ùå Incorrect

```ts
/* eslint @typescript-eslint/consistent-type-definitions: ["error", "type"] */

interface T {
  x: number;
}
```

#### ‚úÖ Correct

```ts
/* eslint @typescript-eslint/consistent-type-definitions: ["error", "type"] */

type T = { x: number };
```

## When Not To Use It

If you specifically want to use an interface or type literal for stylistic reasons, you can disable this rule.
