        <!-- POKERTOOL-HEADER-START
        ---
        schema: pokerheader.v1
project: pokertool
file: pokertool-frontend/node_modules/@typescript-eslint/eslint-plugin/docs/rules/TEMPLATE.md
version: v20.0.0
last_commit: '2025-09-25T18:50:22+01:00'
fixes:
- date: '2025-09-25'
  summary: Enhanced enterprise documentation and comprehensive unit tests added
        ---
        POKERTOOL-HEADER-END -->
> üõë This file is source code, not the primary documentation location! üõë
>
> See **https://typescript-eslint.io/rules/your-rule-name** for documentation.

## Examples

To fill out: tell us more about this rule.

<!--tabs-->

### ‚ùå Incorrect

```ts
// To fill out: incorrect code
```

### ‚úÖ Correct

```ts
// To fill out: correct code
```

## When Not To Use It

To fill out: why wouldn't you want to use this rule?
For example if this rule requires a feature released in a certain TS version.
