// @mui/material/legacy/Link/getTextDecoration.js
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/@mui/material/legacy/Link/getTextDecoration.js
# version: v20.0.0
# last_commit: '2025-09-25T18:50:22+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
import { getPath } from '@mui/system';
import { alpha } from '@mui/system/colorManipulator';
export var colorTransformations = {
  primary: 'primary.main',
  textPrimary: 'text.primary',
  secondary: 'secondary.main',
  textSecondary: 'text.secondary',
  error: 'error.main'
};
var transformDeprecatedColors = function transformDeprecatedColors(color) {
  return colorTransformations[color] || color;
};
var getTextDecoration = function getTextDecoration(_ref) {
  var theme = _ref.theme,
    ownerState = _ref.ownerState;
  var transformedColor = transformDeprecatedColors(ownerState.color);
  var color = getPath(theme, "palette.".concat(transformedColor), false) || ownerState.color;
  var channelColor = getPath(theme, "palette.".concat(transformedColor, "Channel"));
  if ('vars' in theme && channelColor) {
    return "rgba(".concat(channelColor, " / 0.4)");
  }
  return alpha(color, 0.4);
};
export default getTextDecoration;
