// ajv-keywords/src/definitions/typeof.ts
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/schema-utils/node_modules/ajv-keywords/src/definitions/typeof.ts
# version: v28.0.0
# last_commit: '2025-09-25T18:50:22+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
import type {CodeKeywordDefinition, KeywordCxt} from "ajv"
import {_} from "ajv/dist/compile/codegen"

const TYPES = ["undefined", "string", "number", "object", "function", "boolean", "symbol"]

export default function getDef(): CodeKeywordDefinition {
  return {
    keyword: "typeof",
    schemaType: ["string", "array"],
    code(cxt: KeywordCxt) {
      const {data, schema, schemaValue} = cxt
      cxt.fail(
        typeof schema == "string"
          ? _`typeof ${data} != ${schema}`
          : _`${schemaValue}.indexOf(typeof ${data}) < 0`
      )
    },
    metaSchema: {
      anyOf: [
        {type: "string", enum: TYPES},
        {type: "array", items: {type: "string", enum: TYPES}},
      ],
    },
  }
}

module.exports = getDef
