// workbox-build/src/lib/additional-manifest-entries-transform.ts
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/workbox-build/src/lib/additional-manifest-entries-transform.ts
# version: v28.0.0
# last_commit: '2025-09-23T07:20:25+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
/*
  Copyright 2019 Google LLC

  Use of this source code is governed by an MIT-style
  license that can be found in the LICENSE file or at
  https://opensource.org/licenses/MIT.
*/

import {errors} from './errors';
import {ManifestEntry} from '../types';

type AdditionalManifestEntriesTransform = {
  (manifest: Array<ManifestEntry & {size: number}>): {
    manifest: Array<ManifestEntry & {size: number}>;
    warnings: string[];
  };
};

export function additionalManifestEntriesTransform(
  additionalManifestEntries: Array<ManifestEntry | string>,
): AdditionalManifestEntriesTransform {
  return (manifest: Array<ManifestEntry & {size: number}>) => {
    const warnings: Array<string> = [];
    const stringEntries = new Set<string>();

    for (const additionalEntry of additionalManifestEntries) {
      // Warn about either a string or an object that lacks a revision property.
      // (An object with a revision property set to null is okay.)
      if (typeof additionalEntry === 'string') {
        stringEntries.add(additionalEntry);
        manifest.push({
          revision: null,
          size: 0,
          url: additionalEntry,
        });
      } else {
        if (additionalEntry && additionalEntry.revision === undefined) {
          stringEntries.add(additionalEntry.url);
        }
        manifest.push(Object.assign({size: 0}, additionalEntry));
      }
    }

    if (stringEntries.size > 0) {
      let urls = '\n';
      for (const stringEntry of stringEntries) {
        urls += `  - ${stringEntry}\n`;
      }

      warnings.push(errors['string-entry-warning'] + urls);
    }

    return {
      manifest,
      warnings,
    };
  };
}
