// workbox-build/src/lib/get-composite-details.ts
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/workbox-build/src/lib/get-composite-details.ts
# version: v20.0.0
# last_commit: '2025-09-23T07:20:25+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
/*
  Copyright 2018 Google LLC

  Use of this source code is governed by an MIT-style
  license that can be found in the LICENSE file or at
  https://opensource.org/licenses/MIT.
*/

import crypto from 'crypto';

import {FileDetails} from '../types';

export function getCompositeDetails(
  compositeURL: string,
  dependencyDetails: Array<FileDetails>,
): FileDetails {
  let totalSize = 0;
  let compositeHash = '';

  for (const fileDetails of dependencyDetails) {
    totalSize += fileDetails.size;
    compositeHash += fileDetails.hash;
  }

  const md5 = crypto.createHash('md5');
  md5.update(compositeHash);
  const hashOfHashes = md5.digest('hex');

  return {
    file: compositeURL,
    hash: hashOfHashes,
    size: totalSize,
  };
}
