// eslint-plugin-jsx-a11y/__tests__/__util__/parserOptionsMapper.js
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/eslint-plugin-jsx-a11y/__tests__/__util__/parserOptionsMapper.js
# version: v20.0.0
# last_commit: '2025-09-25T18:50:22+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
import { version as eslintVersion } from 'eslint/package.json';
import semver from 'semver';

const usingLegacy = semver.major(eslintVersion) < 9;

const defaultParserOptions = {
  ecmaFeatures: {
    experimentalObjectRestSpread: true,
    jsx: true,
  },
};

const defaultLegacyParserOptions = {
  ...defaultParserOptions,
  ecmaVersion: 2018,
};

const defaultLanguageOptions = {
  ecmaVersion: 'latest',
  parserOptions: {
    ...defaultParserOptions,
  },
};

export default function parserOptionsMapper({
  code,
  errors,
  options = [],
  languageOptions = {},
  settings = {},
}) {
  return usingLegacy
    ? {
      code,
      errors,
      options,
      parserOptions: {
        ...defaultLegacyParserOptions,
        ...languageOptions,
      },
      settings,
    }
    : {
      code,
      errors,
      options,
      languageOptions: {
        ...defaultLanguageOptions,
        ...languageOptions,
      },
      settings,
    };
}
