// @babel/eslint-parser/lib/worker/babel-core.cjs
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/@babel/eslint-parser/lib/worker/babel-core.cjs
# version: v28.0.0
# last_commit: '2025-09-23T07:20:25+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
"use strict";

module.exports = exports;
function initialize(babel) {
  exports.init = null;
  exports.version = babel.version;
  exports.traverse = babel.traverse;
  exports.types = babel.types;
  exports.tokTypes = babel.tokTypes;
  exports.parseSync = babel.parseSync;
  exports.parseAsync = babel.parseAsync;
  exports.loadPartialConfigSync = babel.loadPartialConfigSync;
  exports.loadPartialConfigAsync = babel.loadPartialConfigAsync;
  exports.createConfigItemAsync = babel.createConfigItemAsync;
  {
    exports.createConfigItemSync = babel.createConfigItemSync || babel.createConfigItem;
  }
}
{
  initialize(require("@babel/core"));
}
