// eslint-module-utils/contextCompat.js
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/eslint-module-utils/contextCompat.js
# version: v28.0.0
# last_commit: '2025-09-23T07:20:25+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
'use strict';

exports.__esModule = true;

/** @type {import('./contextCompat').getAncestors} */
function getAncestors(context, node) {
  const sourceCode = getSourceCode(context);

  if (sourceCode && sourceCode.getAncestors) {
    return sourceCode.getAncestors(node);
  }

  return context.getAncestors();
}

/** @type {import('./contextCompat').getDeclaredVariables} */
function getDeclaredVariables(context, node) {
  const sourceCode = getSourceCode(context);

  if (sourceCode && sourceCode.getDeclaredVariables) {
    return sourceCode.getDeclaredVariables(node);
  }

  return context.getDeclaredVariables(node);
}

/** @type {import('./contextCompat').getFilename} */
function getFilename(context) {
  if ('filename' in context) {
    return context.filename;
  }

  return context.getFilename();
}

/** @type {import('./contextCompat').getPhysicalFilename} */
function getPhysicalFilename(context) {
  if (context.getPhysicalFilename) {
    return context.getPhysicalFilename();
  }

  return getFilename(context);
}

/** @type {import('./contextCompat').getScope} */
function getScope(context, node) {
  const sourceCode = getSourceCode(context);

  if (sourceCode && sourceCode.getScope) {
    return sourceCode.getScope(node);
  }

  return context.getScope();
}

/** @type {import('./contextCompat').getSourceCode} */
function getSourceCode(context) {
  if ('sourceCode' in context) {
    return context.sourceCode;
  }

  return context.getSourceCode();
}

module.exports = {
  getAncestors,
  getDeclaredVariables,
  getFilename,
  getPhysicalFilename,
  getScope,
  getSourceCode,
};
