// fork-ts-checker-webpack-plugin/lib/eslint-reporter/issue/EsLintIssueFactory.js
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/fork-ts-checker-webpack-plugin/lib/eslint-reporter/issue/EsLintIssueFactory.js
# version: v28.0.0
# last_commit: '2025-09-23T07:20:25+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function createIssueFromEsLintMessage(filePath, message) {
    let location;
    if (message.line) {
        location = {
            start: {
                line: message.line,
                column: message.column,
            },
            end: {
                line: message.endLine || message.line,
                column: message.endColumn || message.column,
            },
        };
    }
    return {
        origin: 'eslint',
        code: message.ruleId ? String(message.ruleId) : '[unknown]',
        severity: message.severity === 1 ? 'warning' : 'error',
        message: message.message,
        file: filePath,
        location,
    };
}
function createIssuesFromEsLintResults(results) {
    return results.reduce((messages, result) => [
        ...messages,
        ...result.messages.map((message) => createIssueFromEsLintMessage(result.filePath, message)),
    ], []);
}
exports.createIssuesFromEsLintResults = createIssuesFromEsLintResults;
