// eslint-plugin-jest/lib/rules/prefer-called-with.js
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/eslint-plugin-jest/lib/rules/prefer-called-with.js
# version: v20.0.0
# last_commit: '2025-09-23T07:20:25+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _utils = require("./utils");

var _default = (0, _utils.createRule)({
  name: __filename,
  meta: {
    docs: {
      category: 'Best Practices',
      description: 'Suggest using `toBeCalledWith()` or `toHaveBeenCalledWith()`',
      recommended: false
    },
    messages: {
      preferCalledWith: 'Prefer {{name}}With(/* expected args */)'
    },
    type: 'suggestion',
    schema: []
  },
  defaultOptions: [],

  create(context) {
    return {
      CallExpression(node) {
        if (!(0, _utils.isExpectCall)(node)) {
          return;
        }

        const {
          modifier,
          matcher
        } = (0, _utils.parseExpectCall)(node); // Could check resolves/rejects here but not a likely idiom.

        if (matcher && !modifier) {
          if (['toBeCalled', 'toHaveBeenCalled'].includes(matcher.name)) {
            context.report({
              data: {
                name: matcher.name
              },
              // todo: rename to 'matcherName'
              messageId: 'preferCalledWith',
              node: matcher.node.property
            });
          }
        }
      }

    };
  }

});

exports.default = _default;
