// jsdom/lib/jsdom/living/events/MessageEvent-impl.js
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/jsdom/lib/jsdom/living/events/MessageEvent-impl.js
# version: v20.0.0
# last_commit: '2025-09-23T07:20:25+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
"use strict";

const EventImpl = require("./Event-impl").implementation;

const MessageEventInit = require("../generated/MessageEventInit");

class MessageEventImpl extends EventImpl {
  initMessageEvent(type, bubbles, cancelable, data, origin, lastEventId, source, ports) {
    if (this._dispatchFlag) {
      return;
    }

    this.initEvent(type, bubbles, cancelable);
    this.data = data;
    this.origin = origin;
    this.lastEventId = lastEventId;
    this.source = source;
    this.ports = ports;
  }
}
MessageEventImpl.defaultInit = MessageEventInit.convert(undefined);

module.exports = {
  implementation: MessageEventImpl
};
