// @babel/helpers/lib/helpers/importDeferProxy.js
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/@babel/helpers/lib/helpers/importDeferProxy.js
# version: v20.0.0
# last_commit: '2025-09-23T07:20:25+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = _importDeferProxy;
function _importDeferProxy(init) {
  var ns = null;
  var constValue = function (v) {
    return function () {
      return v;
    };
  };
  var proxy = function (run) {
    return function (_target, p, receiver) {
      if (ns === null) ns = init();
      return run(ns, p, receiver);
    };
  };
  return new Proxy({}, {
    defineProperty: constValue(false),
    deleteProperty: constValue(false),
    get: proxy(Reflect.get),
    getOwnPropertyDescriptor: proxy(Reflect.getOwnPropertyDescriptor),
    getPrototypeOf: constValue(null),
    isExtensible: constValue(false),
    has: proxy(Reflect.has),
    ownKeys: proxy(Reflect.ownKeys),
    preventExtensions: constValue(true),
    set: constValue(false),
    setPrototypeOf: constValue(false)
  });
}
