// jsdom/lib/jsdom/living/mutation-observer/MutationRecord-impl.js
# POKERTOOL-HEADER-START
# ---
# schema: pokerheader.v1
# project: pokertool
# file: pokertool-frontend/node_modules/jsdom/lib/jsdom/living/mutation-observer/MutationRecord-impl.js
# version: v28.0.0
# last_commit: '2025-09-23T07:20:25+01:00'
# fixes:
# - date: '2025-09-25'
#   summary: Enhanced enterprise documentation and comprehensive unit tests added
# ---
# POKERTOOL-HEADER-END
"use strict";

const NodeList = require("../generated/NodeList");

// https://dom.spec.whatwg.org/#mutationrecord
class MutationRecordImpl {
  constructor(globalObject, args, privateData) {
    this._globalObject = globalObject;

    this.type = privateData.type;
    this.target = privateData.target;
    this.previousSibling = privateData.previousSibling;
    this.nextSibling = privateData.nextSibling;
    this.attributeName = privateData.attributeName;
    this.attributeNamespace = privateData.attributeNamespace;
    this.oldValue = privateData.oldValue;

    this._addedNodes = privateData.addedNodes;
    this._removedNodes = privateData.removedNodes;
  }

  get addedNodes() {
    return NodeList.createImpl(this._globalObject, [], {
      nodes: this._addedNodes
    });
  }

  get removedNodes() {
    return NodeList.createImpl(this._globalObject, [], {
      nodes: this._removedNodes
    });
  }
}

module.exports = {
  implementation: MutationRecordImpl
};
